### PR TITLE
Add option to clear the terminal by pressing "c"

### PR DESCRIPTION
**Pull request by Levi-Lesches**
## Description

When using commands like `flutter run` and `flutter attach` on the terminal, pressing "c" will now clear the output. This is very helpful for when errors and other outputs clog the screen with irrelevant information. 

## Related Issues

 #38819  

## Tests

I don't know how to add tests for terminal behavior, but the functionality is pretty simple, just print the `clear` character (already defined in some other file). 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by Levi-Lesches** [comment]
Yeah can anyone tell me if I need tests? Or how to write them? It's just a print statement and an extra `case`.

**Review by jonahwilliams — COMMENTED** [review]
Seems like a reasonable approach, but yes you'll absolutely need tests. See some of the existing test cases in the resident_runner test files

File: packages/flutter_tools/lib/src/resident_runner.dart [hunk 1]
```diff
@@ -1052,16 +1052,16 @@ abstract class ResidentRunner {
       commandHelp.P.print();
       commandHelp.a.print();
     }
-    if (flutterDevices.any((FlutterDevice d) => d.device.supportsScreenshot)) {
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/resident_runner.dart]: I think this shouldn't be deleted.

Comment (Levi-Lesches) [reply packages/flutter_tools/lib/src/resident_runner.dart]: It appears twice, just a few lines before this one (sorry, I meant to mention that), but I can put it in another PR if it'll make conflicts.

**Comment by Levi-Lesches** [comment]
Do you happen to know what those tests failures are and how to fix them? They didn't happen when I ran the tests on my computer, and the lines I added should not cause build commands to fail...

File: packages/flutter_tools/lib/src/resident_runner.dart (hunk 1 above)
Comment (zanderso) [reply packages/flutter_tools/lib/src/resident_runner.dart]: Oh, you're right. I wonder how that happened.

**Comment by zanderso** [comment]
This list:
https://github.com/flutter/flutter/blob/6ee8b6546ba4a5b2dbed57a88d88aa074223fccf/packages/flutter_tools/test/general.shard/resident_runner_test.dart#L375
Needs to have the new option added to it.

**Comment by Levi-Lesches** [comment]
Oh thank you -- I was looking for that since I saw it in the tests.

**Comment by Levi-Lesches** [comment]
I'm not quite sure what happened but apparently the tests are broken before I even touched them (apparently `globals` does not actually define `fs`), so I'll try to fix them and hope my changes don't conflict with anyone else's.

**Comment by Levi-Lesches** [comment]
After doing a git pull a LOT is broken in the files I'm trying to modify... so I'll wait for those to be fixed before continuing, since I'm assuming someone else is working on it in another PR.

**Comment by jonahwilliams** [comment]
@Levi-Lesches all of the tool tests are currently green at head. What sorts of errors are you running into with the tests?

**Comment by Levi-Lesches** [comment]
I added one test to see if running `ResidentRunner.clearScreen()` sets `testLogger.statusText` to `""` and it passed, and I added the command to the help section, so is everything good? If not, I can add more, but I'm not sure what else is needed.

**Comment by jonahwilliams** [comment]
@Levi-Lesches definitely don't rebase like that :( . Can you revert that commit please?

**Comment by Levi-Lesches** [comment]
Sorry, I haven't really used Git collaboratively yet. Which commit are you referring to?

**Comment by jonahwilliams** [comment]
A general workflow I recommend:

1. Update upstream: `git fetch upstream`

2. Branch from master: `git checkout upstream/master -b my_feature_branch`

3. Make changes to local branch: `git commit`

4. Prepare for PR: `git push origin my_feature_branch`

To pull in the latest master:

1. `git fetch upstream`

2. `git pull upstream master`

3. `git push origin my_feature_branch`

**Comment by Levi-Lesches** [comment]
I've been doing that, but accidentally created the feature branch *after* I made this PR, so I merged it to my own master (because that's where this PR is referencing). Should I start a new PR from a feature branch instead?

Also, which particular commit where you referring to that rebased weirdly? I didn't explicitly rebase, just merged my master and feature branches.

**Comment by jonahwilliams** [comment]
All of the commits that aren't yours: https://github.com/flutter/flutter/pull/49819/commits

**Comment by Levi-Lesches** [comment]
Hm, yeah those definitely shouldn't be there. I'll try to clean things up -- shoud I submit a new PR so I can have it come from my own feature branch?

**Comment by jonahwilliams** [comment]
In my experience its often easier to start a new PR rather than make git behave...

**Comment by Levi-Lesches** [comment]
Made a new PR in #50034.

**Comment by jonahwilliams** [comment]
Okay, I'll take a look. Closing this one!

**Pull request closed without merging**